cmd/rtrmon/rtrmon.go

```diff
@@ -585,28 +619,55 @@ func NewComparator(c1, c2 *Client) *Comparator {
 	}
 }
 
-func countFirstSeenOnOrBefore(vrps []*VRPJsonSimple, thresholdTimestamp int64) float64 {
+func isDefinitelyVisible(vrp *VRPJsonSimple, thresholdTimestamp int64) bool {
+	return vrp != nil && vrp.Visible && vrp.FirstSeen <= thresholdTimestamp
+}
+
+func isDefinitelyNotVisisble(vrp *VRPJsonSimple, thresholdTimestamp int64) bool {
+	return vrp == nil || (!vrp.Visible && vrp.LastSeen <= thresholdTimestamp)
+}
+
+func countUnmatched(vrps VRPMap, others VRPMap, thresholdTimestamp int64) float64 {
 	count := 0
 
-	for _, vrp := range vrps {
-		if vrp.FirstSeen <= thresholdTimestamp {
-			count++
+	for key, vrp := range vrps {
+		other := others[key]
+		if isDefinitelyVisible(vrp, thresholdTimestamp) {
+			// VRP is definitely visible, count if other side is definitely not visible
+			if isDefinitelyNotVisisble(other, thresholdTimestamp) {
+				count++
+			}
+		} else if isDefinitelyNotVisisble(vrp, thresholdTimestamp) {
+			// VRP is definitely not visible, count if other side is definitely visible
+			if isDefinitelyVisible(other, thresholdTimestamp) {
+				count++
```

Comment:
I assumed we we not counting the symmetric difference, but the one-directional difference.

You can get the symmetric difference by summing over `lhs`/`rhs`?